Client/Frontend/Home/FirefoxHomeViewController.swift

```diff
@@ -595,6 +595,10 @@ extension FirefoxHomeViewController: DataObserverDelegate {
             }
             // Refresh the AS data in the background so we'll have fresh data next time we show.
             self.profile.panelDataObservers.activityStream.refreshIfNeeded(forceTopSites: false)
+            
+            // TODO: How frequently should we *really* be writing here...
+            // Ideally we'd have a cache to check against to skip unnecessary writes.
+            TopSitesHandler.writeTopSites(topSites: result.map { TopSite(url: $0.url, title: $0.title, faviconUrl: $0.icon?.url) })
```

Comment:
Currently we only write to widget top sites model everytime we fetch it from local db.

Suggestion: 
- Update top sites widget model once when you open app
- When someone adds a website to top sites

Reply:
- Star firefox
- Background firefox

If its same website then don't write it.